### PR TITLE
Feature/sort devicon json and add sort check

### LINKED ISSUE #1337: [OTHER] Alphabetically order devicon.json
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### What would you like to share?

Due to some recent merges, the devicon.json file has become unprdered. This should be fixed as soon as possible, and at the very least before the next release.

### Additional information

Alphabetically ordering the file wilø make it easier to find icons, as well as making it less likely to cause conflicts with future updates to it.

### PR DESCRIPTION
**Pull request by Snailedlt**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened

## Features

1. Sort and re-indent `devicon.json`
2. Add check to `check_icon_pr.py` to see if devicon.json is still sorted

**This PR closes #1337, #1327**

## Notes

Please check both if the functionality looks sound, and if the error messages can be improved

### CODE REVIEW

**Comment by Snailedlt** [comment]
The sorting of devicon.json in this PR needs to be redone because devicon.json has been updated since it was last sorted

**Comment by lunatic-fox** [comment]
Code used to sort `devicon.json`
```python
import json
path = 'devicon.json'

with open(path) as file:
  file: list[dict] = json.load(file)
  file.sort(key=lambda e: e['name'])
  file = json.dumps(file, indent=4) + '\n'

with open(path, 'w') as output_file:
  output_file.write(file)
```

**Comment by Snailedlt** [comment]
@lunatic-fox did you check if the check-icon script deemed it as sorted?

File: .github/scripts/check_icon_pr.py [hunk 1]
```diff
@@ -41,7 +46,7 @@ def main():
 
         err_msg = []
         if devicon_err_msg != "":
-            err_msg.append(devicon_err_msg)
+            err_msg += devicon_err_msg
```
Comment (Snailedlt) [inline .github/scripts/check_icon_pr.py:49]: Whoopsie. This should probably use .append like the others :)

**Comment by lunatic-fox** [comment]
> @lunatic-fox did you check if the check-icon script deemed it as sorted?

Not yet, I'll do later some tests in a private repo.

File: .github/scripts/check_icon_pr.py [hunk 2]
```diff
@@ -41,7 +46,7 @@ def main():
 
         err_msg = []
         if devicon_err_msg != "":
```
Comment (Snailedlt) [inline .github/scripts/check_icon_pr.py:48]:
```suggestion
        if devicon_err_msg != []:
```

File: .github/scripts/check_icon_pr.py (hunk 1 above)
Comment (Snailedlt) [reply .github/scripts/check_icon_pr.py:49]: nvm, .append is for strings only, so we can use `+=` or `.extend`, of which I prefer `.extend`
```suggestion
            err_ms.extend(devicon_err_msg)
```

File: .github/scripts/check_icon_pr.py [hunk 3]
```diff
@@ -40,8 +45,8 @@ def main():
             svg_err_msg = check_svgs(svgs)
 
         err_msg = []
-        if devicon_err_msg != "":
-            err_msg.append(devicon_err_msg)
+        if devicon_err_msg != []:
+            err_ms.extend(devicon_err_msg)
```
Comment (lunatic-fox) [inline .github/scripts/check_icon_pr.py:49]: Fix a reference error.
```suggestion
            err_msg.extend(devicon_err_msg)
```

File: .github/scripts/check_icon_pr.py [hunk 4]
```diff
@@ -18,10 +18,15 @@ def main():
     try:
         all_icons = filehandler.get_json_file_content(args.devicon_json_path)
 
+        devicon_err_msg = []
+        #First check if devicon.json is sorted
+        if sorted(all_icons, key=lambda d: d['name']) != all_icons:
+            devicon_err_msg.append("devicon.json is not sorted correctly. Please make sure that your icon is added in the `devicon.json` file at the correct alphabetic position as seen [here](https://github.com/devicons/devicon/wiki/Updating-%60devicon.json%60)")
```
Comment (lunatic-fox) [inline .github/scripts/check_icon_pr.py:24]: Restructure the error message for better visualization.
```suggestion
            devicon_err_msg.append(f"devicon.json is not sorted correctly.\nPlease make sure that your icon is added in the `devicon.json` file at the correct alphabetic position\nas seen here: https://github.com/devicons/devicon/wiki/Updating-%60devicon.json%60")
```

### Preview
preview

**Review by lunatic-fox — CHANGES_REQUESTED** [review]
It'll work properly after fixing the reference error.
### Preview
preview

**Review by lunatic-fox — APPROVED** [review]
- `devicon.json` is up-to-date and sorted with its latest version: https://github.com/devicons/devicon/commit/eb35d73a014618970ee84759b56883bfc33182b0
- Conflicts are resolved.

Looking good! ✔
Thank you for this great feature! 🚀

**Pull request merged**